core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -307,8 +307,9 @@ boolean isInBackoffPeriod() {
    */
   void startBackoff(final Status status) {
     synchronized (lock) {
-      Preconditions.checkState(!inBackoffPeriod);
-      inBackoffPeriod = true;
+      Preconditions.checkState(backoffStatus == null);
+      backoffStatus = Status.UNAVAILABLE.withDescription("Channel in TRANSIENT_FAILURE state")
+          .withCause(status.asRuntimeException());
```

Comment:
This is passed to RPCs when they fail. So the application will end up seeing this.